### PR TITLE
Update for LYWSD03MMC to accept pvvx style custom messages

### PR DESCRIPTION
**Pull request by csiki2**
https://github.com/pvvx/ATC_MiThermometer
2 digit temperature and humidity support.

{
  "id": "A4:C1:38:77:7C:52",
  "name": "ATC_777C52",
  "rssi": -64,
  "distance": 1.795564,
  "servicedata": "527c7738c1a47d09e20c410b4b3704",
  "servicedatauuid": "0x181a",
  "model": "LYWSD03MMC_PVVX",
  "tempc": 24.29,
  "tempf": 75.722,
  "hum": 32.98,
  "batt": 75,
  "volt": 2.881
}

### CODE REVIEW

File: main/ZgatewayBT.ino [hunk 1]
```diff
@@ -1048,6 +1048,7 @@ JsonObject& process_bledata(JsonObject& BLEdata) {
     Log.trace(F("Checking BLE service data validity" CR));
     const char* service_data = (const char*)(BLEdata["servicedata"] | "");
     if (valid_service_data(service_data)) {
+      int service_len = strlen(service_data);
```
Comment (1technophile) [inline main/ZgatewayBT.ino:1051]: Could you use service_len to replace the others strlen(service_data)?

Comment (csiki2) [reply main/ZgatewayBT.ino:1051]: Of course, originally I planned to do it in a separate pull request.
I will update the branch late afternoon.

Comment (1technophile) [reply main/ZgatewayBT.ino:1051]: >  I planned to do it in a separate pull request.

That would be perfect

Comment (csiki2) [reply main/ZgatewayBT.ino:1051]: Ok, do you plan to merge the branch today?
I will create the new branch and pull request after the merge.

**Comment by 1technophile** [comment]
Thanks!

**Pull request merged**